reflection.md

```diff
@@ -2,8 +2,33 @@
 
 ## Which operators were easiest to implement and why?
 
+All of the operators were easier to implement than expected.  Scala seems
+to provide the coder with lots of freedom for choosing operators to define or
+overload.  I would say that the operators ||, ~, and <*> were the easiest to
+implement, since they were direct translations from the classes Union, Concat, and
+Star, respectively.  After this, <+> only involved Concat-ing `this` with 
+`Star(this)`
+
 ## Which operators were most difficult to implement and why?
 
+The most difficult to implement was probably the repetition operator.  The most
+difficult part was interpreting `'3'{7}` as a function of a literal that took an
+integer as an argument literal, and was equivalent to `myExpression(7)`.  After
+that, I had to do some googling before I found out that we were supposed to define
+a function `apply` that would work as parentheses.  Finally, implementing it
+involved recursion, and was much less trivial than the other operators.
+
+Apart from the `apply` operator, I had a difficult time testing the implicit
+conversions with the sbt console.  I wanted interact with the expressions, so I
+opened sbt, typed `console`, and then did `import dsls.regex._`.  With this
+setting, it would properly do the type conversions for `Literal('3') || '4'`, but
+not for `'4' || Literal('3')`.  After much struggle, I resolved this issue (thanks
+to advice from Jeb) by instead using  `import dsls.regex.RegularExpression._`,
+which brought the implicit  conversion into scope. However, I'm not sure why
+`Literal('3') || '4'` underwent the proper conversions.  From this experience, I
```

Comment:
That's definitely very interesting and again that Scala can act strangely. Did it work exactly as expected or not? That may give some indication as to what it was trying to do. I'm not certain, but it may look in the regular expression class, see the `||` operator and try to cast 4 to a regular expression, but not be able to do this otherwise because it would have to implictly cast 4 to a regular expression. Whether or not that's exactly what's happening though it's interesting to see another one of Scala's many intricacies.